### PR TITLE
One port supports multiple neighbors && Fix the failure to update port

### PR DESCRIPTION
**Pull request by chenpiaoping**
1. One port supports multiple neighbors
2. Fix the failure to update port

### CODE REVIEW

**Comment by chenpiaoping** [comment]
@xieus Sorry that the previous PR led to the failure to update the port. This PR will solve the problem.

File: services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/NeighborRepository.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.stream.Collectors;
+
+public class NeighborRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(NeighborRepository.class);
+    private static final String NEIGHBOR_CACHE_NAME_PREFIX = "neighborCache-";
+
+    private ICache<String, PortNeighbors> neighborCache;
+    private CacheFactory cacheFactory;
+
+    public NeighborRepository(CacheFactory cacheFactory) {
```
Comment (xieus) [inline services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/NeighborRepository.java:40]: @chenpiaoping When you get a chance, please update the PM design doc and list the repositories used in Port? It is time for us to sync the doc with the implementation as we are looking into perf optimization.

File: services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/PortRepository.java [hunk 2]
```diff
@@ -261,126 +260,25 @@ public PortNeighbors getPortNeighbors(Object arg) throws CacheException {
...
-                if (subnetPortIds == null) {
-                    subnetPortIds = new SubnetPortIds(subnetId, new HashSet<>(portIds));
-                } else {
-                    subnetPortIds.getPortIds().addAll(portIds);
-                }
-
-                subnetPortIdsCache.put(subnetId, subnetPortIds);
-            }
-
+            neighborRepository.createNeighbors(neighbors);
```
Comment (xieus) [inline services/port_manager/src/main/java/com/futurewei/alcor/portmanager/repo/PortRepository.java:270]: Good refactor.

**Review by xieus — APPROVED** [review]
LGTM.

**Pull request merged**